### PR TITLE
update fps scripts

### PR DESCRIPTION
**Pull request by hhaAndroid**
## Motivation

- For more accurate statistics, it supports repeated calculations to take the average.
- It is more reasonable to set num_worker to 0.
- Support output markdown format.
- Added documentation for the use of benchmark scripts.

--repeat-num=1

model | fps | times_pre_image(ms)
-- | -- | --
atss/atss_r50_fpn_1x_coco.py | x | y

--repeat-num=3

model | fps | mean_fps | times_pre_image(ms) | mean_times_pre_image(ms)
-- | -- | -- | -- | --
atss/atss_r50_fpn_1x_coco.py | x1,x2,x3  | x | y1,y2,y3 | y

### CODE REVIEW

File: .dev_scripts/benchmark_inference_fps.py [hunk 1]
```diff
@@ -19,12 +19,17 @@ def parse_args():
         type=int,
         default=1,
         help='round a number to a given precision in decimal digits')
+    parser.add_argument(
+        '--repeat-num',
+        type=int,
+        default=1,
+        help='num of repeat measure inference speed')
```
Comment (ZwwWayne) [inline .dev_scripts/benchmark_inference_fps.py:26]: number of repeat times of measurement for averaging the results

File: .dev_scripts/benchmark_inference_fps.py [hunk 2]
```diff
@@ -19,12 +19,17 @@ def parse_args():
...
     parser.add_argument(
         '--out', type=str, help='output path of gathered fps to be stored')
     parser.add_argument(
-        '--max-iter', type=int, default=400, help='num of max iter')
+        '--max-iter', type=int, default=2000, help='num of max iter')
```
Comment (ZwwWayne) [inline .dev_scripts/benchmark_inference_fps.py:30]: is 400 long enough？

File: .dev_scripts/benchmark_inference_fps.py [hunk 3]
```diff
@@ -52,9 +57,42 @@ def parse_args():
     return args
 
 
+def format_dict_to_markdown(result_dict):
+    out_strs = [
```
Comment (ZwwWayne) [inline .dev_scripts/benchmark_inference_fps.py:61]: use terminaltables for convenience

File: .dev_scripts/benchmark_inference_fps.py [hunk 4]
```diff
@@ -52,9 +57,42 @@ def parse_args():
...
+            ms_times_pre_image_str = ','.join(
+                [str(s) for s in ms_times_pre_image])
+            out_strs.append(f'| {name} | {fps_str} | {mean_fps} |'
+                            f'{ms_times_pre_image_str} | '
+                            f'{mean_times_pre_image}| \n')
+        else:
+            out_strs.append(f'| {name} | {fps} | {ms_times_pre_image} | \n')
+
+    if is_multiple_results:
+        out_strs[0] = '| model | fps | mean_fps |times_pre_image(ms) ' \
```
Comment (ZwwWayne) [inline .dev_scripts/benchmark_inference_fps.py:85]: Use terminaltable

File: .dev_scripts/benchmark_inference_fps.py (hunk 2 above)
Comment (hhaAndroid) [reply .dev_scripts/benchmark_inference_fps.py:30]: I found that 400 is not enough.

File: tools/analysis_tools/benchmark.py [hunk 5]
```diff
@@ -66,7 +72,7 @@ def measure_inferense_speed(cfg, checkpoint, max_iter, log_interval,
     data_loader = build_dataloader(
         dataset,
         samples_per_gpu=1,
-        workers_per_gpu=cfg.data.workers_per_gpu,
+        workers_per_gpu=0,
```
Comment (RangiLyu) [inline tools/analysis_tools/benchmark.py:75]: Why set workers to 0?

File: tools/analysis_tools/benchmark.py [hunk 6]
```diff
@@ -123,6 +129,40 @@ def measure_inferense_speed(cfg, checkpoint, max_iter, log_interval,
...
+    if repeat_num > 1:
+        fps_list_ = [round(fps, 1) for fps in fps_list]
+        times_pre_image_list_ = [round(1000 / fps, 1) for fps in fps_list]
+        mean_fps_ = sum(fps_list_) / len(fps_list_)
+        mean_times_pre_image_ = sum(times_pre_image_list_) / len(
+            times_pre_image_list_)
+        print(
+            f'Overall fps: {fps_list_}[{mean_fps_:.1f}] img / s, '
+            f'times per image: '
+            f'{times_pre_image_list_}[{mean_times_pre_image_:.1f}] ms / img',
```
Comment (RangiLyu) [inline tools/analysis_tools/benchmark.py:159]: In meta_file, we keep two decimal places, one seems not enough.

File: docs/useful_tools.md [hunk 7]
```diff
@@ -377,10 +377,35 @@ python tools/dataset_converters/cityscapes.py ${CITYSCAPES_PATH} [-h] [--img-dir
...
+python -m torch.distributed.launch --nproc_per_node=1 --master_port=${PORT} tools/analysis_tools/benchmark.py \
+    ${CONFIG} \
+    ${CHECKPOINT} \
+    [--repeat-num ${REPEAT_NUM}] \
+    [--max-iter ${MAX_ITER}] \
+    [--log-interval ${LOG_INTERVAL}] \
+    --launcher pytorch
+```
+
+Examples: Assume that you have already downloaded the `Faster R-CNN` model checkpoint to the directory `checkpoints/`.
```
Comment (RangiLyu) [inline docs/useful_tools.md:400]: Assuming

File: tools/analysis_tools/benchmark.py (hunk 5 above)
Comment (hhaAndroid) [reply tools/analysis_tools/benchmark.py:75]: The test found that when it is not set to 0, the test time will increase and become unstable. And the comparison found that detectron2 is forced to be set to 0.

Comment (RangiLyu) [reply tools/analysis_tools/benchmark.py:75]: Better to add some comments in the code.

**Review by RangiLyu — APPROVED** [review]
LGTM

File: .dev_scripts/benchmark_inference_fps.py [hunk 8]
```diff
@@ -52,9 +58,43 @@ def parse_args():
     return args
 
 
+def format_dict_to_markdown(result_dict):
```
Comment (ZwwWayne) [inline .dev_scripts/benchmark_inference_fps.py:61]: results_to_md

**Review by ZwwWayne — APPROVED** [review]

File: docs/useful_tools.md [hunk 9]
```diff
@@ -377,10 +377,35 @@ python tools/dataset_converters/cityscapes.py ${CITYSCAPES_PATH} [-h] [--img-dir
...
 ```
 
-## Robust Detection Benchmark
+## Benchmark
+
+### Robust Detection Benchmark
 
 `tools/analysis_tools/test_robustness.py` and`tools/analysis_tools/robustness_eval.py`  helps users to evaluate model robustness. The core idea comes from [Benchmarking Robustness in Object Detection: Autonomous Driving when Winter is Coming](https://arxiv.org/abs/1907.07484). For more information how to evaluate models on corrupted images and results for a set of standard models please refer to [robustness_benchmarking.md](robustness_benchmarking.md).
 
+### FPS Benchmark
```
Comment (ZwwWayne) [inline docs/useful_tools.md:386]: Do we need to translate chinese doc?

**Pull request merged**